applications/system-service/systemapi.h

```diff
@@ -89,9 +86,11 @@ class SystemAPI : public APIBase {
        powerOffInhibitors(),
        mutex(),
        touches(),
-       swipeStates() {
+       swipeStates(),
+       swipeLengths() {
         for(short i = Right; i <= Down; i++){
             swipeStates[(SwipeDirection)i] = true;
+            swipeLengths[(SwipeDirection)i] = 30;
```

Comment:
It looks like you aren't restoring this value from settings? This means it will be lost on restarts.